crates/vhost-user-backend/src/backend.rs

```diff
@@ -110,6 +113,35 @@ pub trait VhostUserBackend: Send + Sync {
         vrings: &[Self::Vring],
         thread_id: usize,
     ) -> Result<()>;
+
+    /// Initiate transfer of internal state for the purpose of migration to/from the back-end.
+    ///
+    /// Depending on `direction`, the state should either be saved (i.e. serialized and written to
+    /// `file`) or loaded (i.e. read from `file` and deserialized).  Note that this function must
+    /// not block during transfer, i.e. I/O to/from `file` must be done outside of this function.
+    fn set_device_state_fd(
+        &self,
+        _direction: VhostTransferStateDirection,
+        _phase: VhostTransferStatePhase,
+        _file: File,
+    ) -> Result<Option<File>> {
+        Err(std::io::Error::new(
+            std::io::ErrorKind::Unsupported,
+            "back end does not support state transfer",
+        ))
+    }
+
+    /// After transferring internal state, check for any resulting errors, including potential
+    /// deserialization errors when loading state.
+    ///
+    /// Although this function return a `Result`, the front-end will not receive any details about
+    /// this error.
+    fn check_device_state(&self) -> Result<()> {
```

Comment:
Personally, I liked using `std::result::Result<(), ()>` to make it explicit in typing that the error message would be lost, because the function/message name sounds like the front-end might want to inquire about the exact nature of the error (and documentation is easy to miss).
But then again, most messages allow you to return a non-trivial error, and I don’t think any of them will return anything more than “error/success” to the front-end, so really, there is no point in treating this any differently from those other messages.

Reply:
Nop, it's a clippy warn:
https://rust-lang.github.io/rust-clippy/master/index.html#/result_unit_err

Reply:
Interesting, for some reason I couldn’t reproduce that warning (and didn’t find anything about it). Ah well.